bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/inbox/InboxFilterCriteria.java

```diff
@@ -26,61 +27,137 @@
     private final String bindingId;
     private final ThingTypeUID thingTypeUID;
     private final ThingUID thingUID;
+    private final String representationValue;
```

Comment:
Well, [below](https://github.com/eclipse/smarthome/pull/3899/files#diff-165ec77b918d80c4f727ab179ac5f93eR50) you add MANY new public methods that become API - which will make it difficult to remove them again. I am ok with keeping backward compatibility by not touching the InboxFilterCriteria here at all. But instead of adding new features to it (filtering for properties), my suggestion is to introduce directly the new stream mechanism for these purposes (especially as I do not think that it will mean much more effort).